hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/protocol/StorageContainerDatanodeProtocol.java

```diff
@@ -84,4 +86,19 @@ SCMRegisteredResponseProto register(
           ContainerReportsProto containerReportsRequestProto,
           PipelineReportsProto pipelineReports) throws IOException;
 
+  /**
+   * Register Datanode.
+   * @param datanodeDetails - Datanode Details.
+   * @param nodeReport - Node Report.
+   * @param containerReportsRequestProto - Container Reports.
+   * @param layoutInfo - Layout Version Information.
+   * @return SCM Command.
+   */
+  SCMRegisteredResponseProto register(
```

Comment:
Why do we need to create a new 'register' method with Layout Version arg, without changing the existing method? We would always need a Layout Version during registering.

Reply:
yup. Done.